### PR TITLE
Add mainstreethub parent pom and enable builds

### PR DESCRIPTION
**Pull request by bentonporter**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
+  <parent>
+    <groupId>com.mainstreethub</groupId>
+    <artifactId>parent-pom</artifactId>
+    <version>1.6.0</version>
+  </parent>
+
   <groupId>net.rcarz</groupId>
   <artifactId>jira-client</artifactId>
-  <version>0.6-SNAPSHOT</version>
+  <version>0.5-1-SNAPSHOT</version>
```
Comment (bbeck) [inline pom.xml]: I'd make this `0.5.0-1`

File: pom.xml [hunk 2]
```diff
@@ -19,8 +24,9 @@
   </licenses>
 
   <scm>
-    <url>https://github.com/rcarz/jira-client</url>
-    <connection>https://github.com/rcarz/jira-client</connection>
+    <url>https://github.com/mainstreethub/jira-client</url>
```
Comment (bbeck) [inline pom.xml]: The SCM section is not needed with the way our CI for library builds works.  It doesn't use the maven-release-plugin which is what reads this section of the POM.

File: pom.xml [hunk 3]
```diff
@@ -93,4 +93,52 @@
     </dependency>
 
   </dependencies>
+
+  <build>
+    <plugins>
+
+      <!-- Override parent pom to suppress enforcing dependency convergence -->
```
Comment (bbeck) [inline pom.xml]: Dependency convergence errors are real problems with your library that could result in runtime errors depending how maven resolves them.  I wouldn't suppress them, instead I'd fix them.

File: pom.xml [hunk 4]
```diff
@@ -93,4 +93,52 @@
...
+                  <version>${java.minimum.version}</version>
+                </requireJavaVersion>
+              </rules>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <!-- Override parent pom to suppress treating warnings as errors -->
+      <plugin>
```
Comment (bbeck) [inline pom.xml]: Depending on these you might just want to fix them -- they are the compiler telling you that things could be wrong.

File: pom.xml (hunk 3 above)
Comment (bentonporter) [reply pom.xml]: I should be able to fix these by excluding the sub-dependencies that conflict and then explicitly declaring them in the pom.

File: pom.xml (hunk 4 above)
Comment (bentonporter) [reply pom.xml]: That's a good point. I started going down this path, but there are a lot of changes needed to clear up the warnings. I'm concerned that if we diverge too much from the original repo that it'll be harder to keep it up-to-date.

**Pull request merged**